### PR TITLE
Using new internal status constants in tests

### PR DESCRIPTION
**Pull request by wjrosa**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Base PR https://github.com/woocommerce/woocommerce/pull/52427.

This PR uses the legacy status constants introduced in https://github.com/woocommerce/woocommerce/pull/52427 in existing unit tests.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Since this changes only unit tests, code review should be enough. Check if the tests are still passing.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Uses the newly introduced legacy order status constants in unit tests.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by barryhughes — COMMENTED** [review]
Looks good! Running tests locally since our workflow isn't doing it. In the meantime, just flagging some other spots we might want to update:

```
$ grep --recursive --line-number -E "wc-(cancelled|completed|failed|on-hold|pending|processing|refunded)" tests/legacy

tests/legacy/unit-tests/admin/reports/class-wc-tests-admin-report.php:149:        $this->assertEquals( 'wc-completed', $data->post_status );
tests/legacy/unit-tests/order/class-wc-tests-order-functions.php:93:              array( 'wc-on-hold', 2 ),
tests/legacy/unit-tests/order/class-wc-tests-order-functions.php:97:              array( 'wc-on-hold', 3 ),
tests/legacy/unit-tests/customer/crud.php:268:                                    $order->update_status( 'wc-completed' );
```

**Comment by wjrosa** [comment]
> Looks good! Running tests locally since our workflow isn't doing it. In the meantime, just flagging some other spots we might want to update:
> 
> ```
> $ grep --recursive --line-number -E "wc-(cancelled|completed|failed|on-hold|pending|processing|refunded)" tests/legacy
> 
> tests/legacy/unit-tests/admin/reports/class-wc-tests-admin-report.php:149:        $this->assertEquals( 'wc-completed', $data->post_status );
> tests/legacy/unit-tests/order/class-wc-tests-order-functions.php:93:              array( 'wc-on-hold', 2 ),
> tests/legacy/unit-tests/order/class-wc-tests-order-functions.php:97:              array( 'wc-on-hold', 3 ),
> tests/legacy/unit-tests/customer/crud.php:268:                                    $order->update_status( 'wc-completed' );
> ```

Thanks, Barry! I have also replaced the missing occurrences in https://github.com/woocommerce/woocommerce/pull/52433/commits/b4464a2d4875c3ef80c40daa27341c696cce032a

File: plugins/woocommerce/tests/php/includes/data-stores/class-wc-customer-data-store-test.php
Comment (kalessil) [inline plugins/woocommerce/tests/php/includes/data-stores/class-wc-customer-data-store-test.php]: I guess `'pending'` value here could be replaced with `OrderStatus::PENDING`?

**Review by kalessil — COMMENTED** [review]
LGTM to me overall!

Leaving one comment to look into. We also need to fix code style violations reported by CI and get @barryhughes approval before I can merge the PR.

File: plugins/woocommerce/tests/php/includes/data-stores/class-wc-customer-data-store-test.php
Comment (wjrosa) [reply plugins/woocommerce/tests/php/includes/data-stores/class-wc-customer-data-store-test.php]: Good catch, @kalessil ! Replaced it and also fixed all lint issues 👍

**Review by kalessil — APPROVED** [review]
LGTM!

**Pull request merged**
